charts/sda-svc/templates/syncapi-deploy.yaml

```diff
@@ -164,15 +169,15 @@ spec:
       volumes:
       {{- if and (not .Values.global.pkiService) .Values.global.tls.enabled }}
         - name: tls
-          projected:
-            sources:
-          {{- if or .Values.global.tls.clusterIssuer .Values.global.tls.issuer }}
-            - secret:
-                name: {{ template "sda.fullname" . }}-syncapi-certs
-          {{- else }}
-            - secret:
-                name: {{ required "An certificate issuer or a TLS secret name is required for syncapi" .Values.syncapi.tls.secretName }}
-          {{- end }}
+        {{- if or .Values.global.tls.clusterIssuer .Values.global.tls.issuer }}
+          secret:
+            defaultMode: 0440
+            secretName: {{ template "sda.fullname" . }}-syncapi-certs
+        {{- else if .Values.syncapi.tls.secretName }}
+          secret:
+            defaultMode: 0440
+            secretName: {{ required "An certificate issuer or a TLS secret name is required for syncapi" .Values.syncapi.tls.secretName }}
```

Comment:
Since we're checking for `.Values.syncapi.tls.secretName` in the helm conditional, the `required doesn't do anything here?

Reply:
Rather the opposite since the `if else` isn't a bool check it gets ignored

Reply:
I don't think I understand what you mean here - the helm flow control introduction (https://helm.sh/docs/chart_template_guide/control_structures/#ifelse) specifically lists the sets of values for the pipeline that are considered as false. Of course, `required` isn't the same but I can't think of any cases where `required` will trigger an error here.

But this should be harmless and not a hindrance for merge.

Reply:
`if .Values.syncapi.tls.secretName` will only trigger if `if .Values.syncapi.tls.secretName` is unset or bool `false`, not if it is an empty string. (which is the default in the values file)

Reply:
That sounds odd, is this some non-standard behaviour activated somewhere?

This shows what I expect, which I think aligns with the helm documentation:

```
pontus@lalage:/tmp$ cat testdir/templates/testfil 
# Testing a few evaluations

{{ if "" }}
# Empty string is true
{{ end }}

{{ if 0 }}
# 0 is true
{{ end }}

{{ if .Values.DoesNotExist }}
# {{ .Values.DoesNotExist }} from unset is true
{{ end }}

{{ if false }}
# false is true
{{ end }}

pontus@lalage:/tmp$ helm template testdir
---
# Source: testchart/templates/testfil
# Testing a few evaluations
pontus@lalage:/tmp$ 
```